### PR TITLE
[AIRFLOW-6353] security - ui - add click jacking defence

### PR DESCRIPTION
**Pull request by tooptoop4**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-6353\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6353
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-6353\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by potiuk** [comment]
Should not we make it optional and controlled by a parameter (on by default). I can very easily imagine some service vendors might want to embed Apache Airflow in their own pre-existing interface (for example when you want to expose Airflow as some kind of service within a bigger context). Not sure if this is happening currently but I can imagine it can.

What do you think @tooptoop4 ?

**Comment by tooptoop4** [comment]
@potiuk added config

**Comment by tooptoop4** [comment]
not sure if this CI test fail is relevant:

=================================== FAILURES ===================================
____ TestKubernetesExecutor.test_integration_run_dag_with_scheduler_failure ____
self = <tests.integration.kubernetes.test_kubernetes_executor.TestKubernetesExecutor testMethod=test_integration_run_dag_with_scheduler_failure>
    def test_integration_run_dag_with_scheduler_failure(self):
        host = KUBERNETES_HOST
        dag_id = 'example_kubernetes_executor_config'
    
        result_json = self.start_dag(dag_id=dag_id, host=host)
    
        self.assertGreater(len(result_json['items']), 0)
    
        execution_date = result_json['items'][0]['execution_date']
        print("Found the job with execution date {}".format(execution_date))
    
        self._delete_airflow_pod()
    
        time.sleep(10)  # give time for pod to restart
    
        # Wait 100 seconds for the operator to complete
        self.monitor_task(host=host,
                          execution_date=execution_date,
                          dag_id=dag_id,
                          task_id='start_task',
>                         expected_final_state='success', timeout=120)
tests/integration/kubernetes/test_kubernetes_executor.py:225: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/integration/kubernetes/test_kubernetes_executor.py:105: in monitor_task
    self.assertEqual(state, expected_final_state)
E   AssertionError: 'failed' != 'success'
E   - failed
E   + success
----------------------------- Captured stdout call -----------------------------
Found the job with execution date 2019-12-27T19:34:49+00:00
pod "airflow-5757bbd59f-msr6q" deleted
[2019-12-27 19:35:16,105] {connectionpool.py:745} WARNING - Retrying ('\x1b[1mRetry(total=2, connect=None, read=None, redirect=None, status=None)\x1b[0m') after connection broken by ''\x1b[1m<urllib3.connection.HTTPConnection object at 0x7f34e0b100b8>: Failed to establish a new connection: [Errno 111] Connection refused\x1b[0m'': /api/experimental/dags/example_kubernetes_executor_config/dag_runs/2019-12-27T19:34:49+00:00/tasks/start_task
[2019-12-27 19:35:18,110] {connectionpool.py:745} WARNING - Retrying ('\x1b[1mRetry(total=1, connect=None, read=None, redirect=None, status=None)\x1b[0m') after connection broken by ''\x1b[1m<urllib3.connection.HTTPConnection object at 0x7f34e0b2e1d0>: Failed to establish a new connection: [Errno 111] Connection refused\x1b[0m'': /api/experimental/dags/example_kubernetes_executor_config/dag_runs/2019-12-27T19:34:49+00:00/tasks/start_task
Attempt 0: Current state of operator is failed
Attempt 1: Current state of operator is failed
Attempt 2: Current state of operator is failed
Attempt 3: Current state of operator is failed
Attempt 4: Current state of operator is failed
Attempt 5: Current state of operator is failed
Attempt 6: Current state of operator is failed
Attempt 7: Current state of operator is failed
Attempt 8: Current state of operator is failed
Attempt 9: Current state of operator is failed
Attempt 10: Current state of operator is failed
Attempt 11: Current state of operator is failed
Attempt 12: Current state of operator is failed
Attempt 13: Current state of operator is failed
Attempt 14: Current state of operator is failed
Attempt 15: Current state of operator is failed
Attempt 16: Current state of operator is failed
Attempt 17: Current state of operator is failed
Attempt 18: Current state of operator is failed
Attempt 19: Current state of operator is failed
Attempt 20: Current state of operator is failed
Attempt 21: Current state of operator is failed
Attempt 22: Current state of operator is failed
Attempt 23: Current state of operator is failed
------------------------------ Captured log call -------------------------------
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying ('\x1b[1mRetry(total=2, connect=None, read=None, redirect=None, status=None)\x1b[0m') after connection broken by ''\x1b[1m<urllib3.connection.HTTPConnection object at 0x7f34e0b100b8>: Failed to establish a new connection: [Errno 111] Connection refused\x1b[0m'': /api/experimental/dags/example_kubernetes_executor_config/dag_runs/2019-12-27T19:34:49+00:00/tasks/start_task
WARNING  urllib3.connectionpool:connectionpool.py:745 Retrying ('\x1b[1mRetry(total=1, connect=None, read=None, redirect=None, status=None)\x1b[0m') after connection broken by ''\x1b[1m<urllib3.connection.HTTPConnection object at 0x7f34e0b2e1d0>: Failed to establish a new connection: [Errno 111] Connection refused\x1b[0m'': /api/experimental/dags/example_kubernetes_executor_config/dag_runs/2019-12-27T19:34:49+00:00/tasks/start_task
=============================== warnings summary ===============================
/usr/local/lib/python3.6/site-packages/_pytest/junitxml.py:436
  /usr/local/lib/python3.6/site-packages/_pytest/junitxml.py:436: PytestDeprecationWarning: The 'junit_family' default value will change to 'xunit2' in pytest 6.0.
  Add 'junit_family=legacy' to your pytest.ini file to silence this warning and make your suite compatible.
    _issue_warning_captured(deprecated.JUNIT_XML_DEFAULT_FAMILY, config.hook, 2)
-- Docs: https://docs.pytest.org/en/latest/warnings.html
------------- generated xml file: /root/airflow/logs/all_tests.xml -------------
---------- coverage: platform linux, python 3.6.10-final-0 -----------
Coverage HTML written to dir airflow/www/static/coverage/
========================== slowest 100 test durations ==========================
158.67s call     tests/integration/kubernetes/test_kubernetes_executor.py::TestKubernetesExecutor::test_integration_run_dag_with_scheduler_failure

**Comment by potiuk** [comment]
Conflicts now = needs to be fixed :(

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -362,6 +362,9 @@ cookie_samesite =
 # Default setting for wrap toggle on DAG code and TI log views.
 default_wrap = False
 
+# Allow the UI to be rendered in a frame
```
Comment (mik-laj) [inline airflow/config_templates/default_airflow.cfg:365]: Hi. Can you add some documentation? I think that a short description in the security.rst file was enough. Very few people will use this feature without documentation.

Comment (tooptoop4) [reply airflow/config_templates/default_airflow.cfg:365]: see https://github.com/apache/airflow/pull/6995

**Pull request closed without merging**